packages/editor/src/components/visual-editor/index.js

```diff
@@ -406,6 +406,9 @@ function VisualEditor( {
 							...deviceStyles,
 						},
 					} }
+					// When there’s no iframe the canvas is the scrolling context and with the
+					// iframe, device previews may overflow vertically.
+					enableScroll={ disableIframe || deviceType !== 'Desktop' }
```

Comment:
I'm wondering if there's a way to solve this at a higher level without adding a new prop to the hoge component. Here's a rough diff, but what do you think?

```diff
diff --git a/packages/block-editor/src/components/block-canvas/index.js b/packages/block-editor/src/components/block-canvas/index.js
index ef70c4a26d..dd24a91970 100644
--- a/packages/block-editor/src/components/block-canvas/index.js
+++ b/packages/block-editor/src/components/block-canvas/index.js
@@ -1,8 +1,3 @@
-/**
- * External dependencies
- */
-import clsx from 'clsx';
-
 /**
  * WordPress dependencies
  */
@@ -38,7 +33,6 @@ export function ExperimentalBlockCanvas( {
        styles,
        contentRef: contentRefProp,
        iframeProps,
-       enableScroll,
 } ) {
        useBlockCommands();
        const isTabletViewport = useViewportMatch( 'medium', '<' );
@@ -57,18 +51,10 @@ export function ExperimentalBlockCanvas( {
                                        frameSize: '40px',
                          }
                        : {};
-       const className = clsx(
-               'block-editor-block-canvas',
-               enableScroll && 'is-scrollable'
-       );
 
        if ( ! shouldIframe ) {
                return (
-                       <BlockTools
-                               __unstableContentRef={ localRef }
-                               className={ className }
-                               style={ { height } }
-                       >
+                       <BlockTools __unstableContentRef={ localRef } style={ { height } }>
                                <EditorStyles
                                        styles={ styles }
                                        scope=":where(.editor-styles-wrapper)"
@@ -88,7 +74,6 @@ export function ExperimentalBlockCanvas( {
        return (
                <BlockTools
                        __unstableContentRef={ localRef }
-                       className={ className }
                        style={ { height, display: 'flex' } }
                >
                        <Iframe
diff --git a/packages/block-editor/src/components/block-canvas/style.scss b/packages/block-editor/src/components/block-canva
s/style.scss
index 52c204407e..840ef3644c 100644
--- a/packages/block-editor/src/components/block-canvas/style.scss
+++ b/packages/block-editor/src/components/block-canvas/style.scss
@@ -1,4 +1,4 @@
-.block-editor-block-canvas.is-scrollable {
+.edit-post-visual-editor.is-scrollable {
        overflow: auto;
 
        // Applicable only when legacy (non-iframed).
diff --git a/packages/editor/src/components/visual-editor/index.js b/packages/editor/src/components/visual-editor/index.js
index 5a75355e08..122252ea4a 100644
--- a/packages/editor/src/components/visual-editor/index.js
+++ b/packages/editor/src/components/visual-editor/index.js
@@ -385,6 +385,7 @@ function VisualEditor( {
                                        'has-padding': isFocusedEntity || enableResizing,
                                        'is-resizable': enableResizing,
                                        'is-iframed': ! disableIframe,
+                                       'is-scrollable': disableIframe || deviceType !== 'Desktop',
                                }
                        ) }
                >
@@ -406,9 +407,6 @@ function VisualEditor( {
                                                        ...deviceStyles,
                                                },
                                        } }
-                                       // When there’s no iframe the canvas is the scrolling context and with the
-                                       // iframe, device previews may overflow vertically.
-                                       enableScroll={ disableIframe || deviceType !== 'Desktop' }
                                >
                                        { themeSupportsLayout &&
                                                ! themeHasDisabledLayoutStyles &&
```

Reply:
Thanks for suggesting this. I’ve tried it. The only thing is actually having the visual editor’s root element as scrollable is trickier as it won’t contain the margins of the device previews. That’s because the block canvas’ root element has an inline-styled height of `100%` (as specified by VisualEditor) and that’s important for fluid transitions to/from device previews. So to get the margins to be included in the overflow the height has to be `auto` while scrollable but then it breaks the transitions to a device preview.

<details><summary>Screen recording to demonstrate issues with <code>VisualEditor</code> as the scrolling context</summary>

https://github.com/user-attachments/assets/9b84de69-c2c3-4293-8e06-7e5e0f71fee2

</details>

Still, it looks like we can avoid adding a prop to `ExperimentalBlockCanvas` d00892256f310540bb806e8b507c59332f99c6b9. The component still needs a class added to target through CSS. It makes for what may be considered slightly worse style encapsulation because now the editor package is relying on a class applied in the block-editor package. For that reason, I think adding the prop may be slighter "cleaner" 🤷.